client-side-js+browser-api/task7/main.js

```diff
@@ -0,0 +1,10 @@
+const highlightButton = document.getElementById("highlight-button");
+
+const highlightParagraphs = () => {
+  const paragraphs = document.querySelectorAll("p");
+  paragraphs.forEach((p) => {
```

Comment:
Good job!

A few optimization tips that do not affect the score of the task:
1. If we know that there is only static content on the page. 
In this case we don't need to search for 'p' elements every time we call the callback function. If you move the search for 'p' elements from line 4 to line 2, you can save some resources.

2. If you need to search for 'p' tags every callback function call. 
In this callback I see that the variable 'paragraphs' is not reused, and you can save some memory by calling the forEach method directly from the querySelector: document.querySelectorAll("p").forEach(...).

Reply:
fixed